lib/new_relic/agent/log_event_aggregator.rb

```diff
@@ -172,6 +172,7 @@ def after_harvest metadata
       # To avoid paying the cost of metric recording on every line, we hold
       # these until harvest before recording them
       def record_customer_metrics
+        return unless NewRelic::Agent.config[:'application_logging.metrics.enabled']
```

Comment:
I wonder if we should be gating the incrementing in `record` as well? Otherwise, we'll keep counting and eventually the integer might trip to a big int 🤔  Doesn't _seem_ like it should be a real problem, but unnecessary work we're doing there at the least.

Reply:
Great point! I'll add that to the draft PR for #1007